operators/multiclusterobservability/controllers/placementrule/predicate_func.go

```diff
@@ -100,18 +100,19 @@ func getClusterPreds() predicate.Funcs {
 	}
 }
 
-func GetAddOnDeploymentPredicates() predicate.Funcs {
+func GetAddOnDeploymentConfigPredicates() predicate.Funcs {
 	return predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
 			return true
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			if !reflect.DeepEqual(e.ObjectNew.(*addonv1alpha1.AddOnDeploymentConfig).Spec.ProxyConfig,
-				e.ObjectOld.(*addonv1alpha1.AddOnDeploymentConfig).Spec.ProxyConfig) {
-				log.Info("AddonDeploymentConfig is updated", e.ObjectNew.GetName(), "name", e.ObjectNew.GetNamespace(), "namespace")
-				return true
+			newObj := e.ObjectNew.(*addonv1alpha1.AddOnDeploymentConfig)
+			oldObj := e.ObjectOld.(*addonv1alpha1.AddOnDeploymentConfig)
+			if reflect.DeepEqual(newObj, oldObj) {
```

Comment:
I wonder would it be better here to just do a DeepEqual check on the entire Spec as opposed to the entire resource.  
I am considering the possibility here where you get into an endless loop due to checking things like updatedAt timestamps and revision numbers when all we really care about is changes to the Spec itself?

Reply:
Good one, @philipgough. Fixed in https://github.com/stolostron/multicluster-observability-operator/pull/1376/commits/8429c0d9a21c654e90ad75dacab57a163a31b1fb